frontend/app/controllers/spree/checkout_controller.rb

```diff
@@ -165,6 +165,23 @@ def completion_route
       spree.order_path(@order)
     end
 
+    def apply_coupon_code
+      if params[:order] && params[:order][:coupon_code].present?
+        @order.coupon_code = params[:order][:coupon_code]
```

Comment:
We should use `params.require(:order).permit(:coupon_code)` instead. This is more secure and will most likely fail  in Rails >= 5.1 otherwise.